engine/access/rpc/backend/backend.go

```diff
@@ -179,3 +191,38 @@ func convertStorageError(err error) error {
 
 	return status.Errorf(codes.Internal, "failed to find: %v", err)
 }
+
+// executionNodesForBlockID returns upto maxExecutionNodesCnt number of randomly chosen execution node identities
+// which have executed the given block ID. If no such execution node is found, then an error is returned.
+func executionNodesForBlockID(
```

Comment:
this is a common method called by any `backend_*` struct that needs to access the Execution node